x-pack/plugins/lens/public/xy_visualization/visualization.tsx

```diff
@@ -287,8 +289,94 @@ export const xyVisualization: Visualization<State> = {
 
   toExpression,
   toPreviewExpression,
+
+  getErrorMessages(state, frame) {
+    // Data error handling below here
+    const hasNoXAccessor = ({ xAccessor }: LayerConfig) => xAccessor == null;
+    const hasNoAccessors = ({ accessors }: LayerConfig) =>
+      accessors == null || accessors.length === 0;
+
+    const errors: Array<{
+      shortMessage: string;
+      longMessage: string;
+    }> = [];
+
+    // check if the layers in the state are compatible with this type of chart
+    if (state && state.layers.length > 1) {
+      const checks: Array<[string, (layer: LayerConfig) => boolean]> = [
+        ['X', hasNoXAccessor],
```

Comment:
For percentage charts, it's the breakdown dimension which is required instead of the x axis. This leads to the following weird state: workspace message says x axis is missing, config panel says breakdown by is missing:
<img width="864" alt="Screenshot 2020-10-29 at 10 45 50" src="https://user-images.githubusercontent.com/1508364/97552200-42d5c880-19d4-11eb-91dd-6bfe68fb237a.png">

Reply:
Yes, I noticed that too this morning. Already working on it 👍